File: app/src/main/java/org/apache/fineract/data/services/DepositService.java
package org.apache.fineract.data.services;

import org.apache.fineract.data.models.deposit.DepositAccount;
import org.apache.fineract.data.models.deposit.ProductDefinition;
import org.apache.fineract.data.remote.EndPoints;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * @author Rajan Maurya
 *         On 07/07/17.
 */

public interface DepositService {

    @GET(EndPoints.API_DEPOSIT_PATH + "/instances")
    Observable<List<DepositAccount>> fetchCustomersDeposits(
            @Query("customer") String customerIdentifier);

    @GET(EndPoints.API_DEPOSIT_PATH + "/instances/{accountIdentifier}")
    Observable<DepositAccount> fetchCustomerDepositDetails(
            @Path("accountIdentifier") String accountIdentifier);

    @GET(EndPoints.API_DEPOSIT_PATH + "/definitions")
    Observable<List<ProductDefinition>> fetchProductDefinitions();

    @POST(EndPoints.API_DEPOSIT_PATH + "/instances")
    Completable createDepositAccount(@Body DepositAccount depositAccount);

    @PUT(EndPoints.API_DEPOSIT_PATH + "/instances/{accountIdentifier}")
    Completable updateDepositAccount(
            @Path("accountIdentifier") String accountIdentifier,
            @Body DepositAccount depositAccount);
}

File: app/src/main/java/org/apache/fineract/data/services/DepositService.kt
package org.apache.fineract.data.services

import io.reactivex.Completable
import io.reactivex.Observable
import org.apache.fineract.data.models.deposit.DepositAccount
import org.apache.fineract.data.models.deposit.ProductDefinition
import org.apache.fineract.data.remote.EndPoints
import retrofit2.http.*

/**
 * Created by Ahmad Jawid Muhammadi on 2/6/20
 */
interface DepositService {
    @GET(EndPoints.API_DEPOSIT_PATH + "/instances")
    fun fetchCustomersDeposits(
            @Query("customer") customerIdentifier: String?): Observable<List<DepositAccount?>?>?

    @GET(EndPoints.API_DEPOSIT_PATH + "/instances/{accountIdentifier}")
    fun fetchCustomerDepositDetails(
            @Path("accountIdentifier") accountIdentifier: String?): Observable<DepositAccount?>?

    @GET(EndPoints.API_DEPOSIT_PATH + "/definitions")
    fun fetchProductDefinitions(): Observable<List<ProductDefinition?>?>?

    @POST(EndPoints.API_DEPOSIT_PATH + "/instances")
    fun createDepositAccount(@Body depositAccount: DepositAccount?): Completable?

    @PUT(EndPoints.API_DEPOSIT_PATH + "/instances/{accountIdentifier}")
    fun updateDepositAccount(
            @Path("accountIdentifier") accountIdentifier: String?,
            @Body depositAccount: DepositAccount?): Completable?
}